sdk/servicebus/service-bus/test/public/atomManagement.spec.ts

```diff
@@ -2638,12 +2638,17 @@ versionsToTest(serviceApiVersions, {}, (serviceVersion, {}) => {
               "TestError: Topic path AND subscription path must be passed when invoking tests on rules"
             );
           }
+          if (!ruleOptions?.filter || !ruleOptions.action) {
+            throw new Error(
+              "TestError: ruleOptions.filter and ruleOptions.action should have been set"
```

Comment:
Now the live tests actually failed because `ruleOptions` is `undefined`.  Before this PR we just force-passing `undefined` for filter and action to `atomClient.createRule()`.  I also see a discrepancy that while `createRule` doesn't allow `undefined` values, they are allowed in `putResource` (as `CreateRuleOptions`): https://github.com/Azure/azure-sdk-for-js/blob/16b61476ae2b2f82c916160ef8dfb53a13ddcccc/sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts#L2198  

Any insights? @HarshaNalluru @ramya-rao-a 
Another option is to ignore the linting error if this is indeed expected.

Reply:
There is no point in creating a rule without a filter right?

Reply:
But you can create a rule without action.
We have overloads :)
![image](https://user-images.githubusercontent.com/10452642/142719969-a897bd83-4e90-4c78-a60c-1a3bfd30b7be.png)

Reply:
> There is no point in creating a rule without a filter, right?
Then I don't know why the test would pass `undefined` filter to `createRule()`.  We do create a always true sql filter internally

Reply:
Hmmm, we can remove the test

Reply:
any idea/background on the reason behind creating always true filter?

Reply:
No background as such, we just didn't remove it when we migrated.
We don't need to do it I think, I believe the service adds the true filter anyways. Can you check and remove maybe?

Reply:
In this test file, we derive the type of `ruleOptions` from `CreateSubscriptionOptions.defaultRuleOptions` so `filter` is optional. I am going to ignore linting for this line since it's just testing, and the test passed.

Reply:
https://github.com/Azure/azure-sdk-for-js/blob/9f454f84a636b6a511fa28d81496f0e6573b13d5/sdk/servicebus/service-bus/test/public/atomManagement.spec.ts#L2569

Reply:
hmm, `filter` is actually required now because of `Required<>`.  undefined is passed because `ruleOptions` is undefined.

Reply:
So how do you want to move forward?
- make the rule options required
  - in that, filter will be required
  - action is optional
  - check for the existence of filter and throw if it is not provided to avoid `!` at the end
  - check for the presence of action and call the overloads accordingly
  - update/remove the tests as needed

Reply:
The API we use look fine. I had added throw will update it to only check filter. Don't need to do anything for action since it's optional, right?  I will remove the test scenario of undefined ruleOptions as it doesn't apply for `createRule()`

Reply:
yes, you are right: calling different overloads based on action value